project/api/write.py

```diff
@@ -254,46 +251,63 @@ def deleteCivi(request):
 
 @login_required
 def editThread(request):
-
     thread_id = request.POST.get('thread_id')
+
+    if not thread_id:
+        return HttpResponseBadRequest(reason="Invalid Thread Reference")
+
+    # Change Publish State to True
+    is_draft = request.POST.get('is_draft', True)
+
+    if not is_draft:
+        try:
+            draft_thread = Thread.objects.get(id=thread_id)
+        except Exception as e:
+            return HttpResponseServerError(reason=str(e))
+
+        draft_thread.is_draft = False
+
+        # try:
+        #     draft_thread.save()
+        # except Exception as e:
+        #     return HttpResponseServerError(reason=str(e))
+
+        return JsonResponse({'data': "Success"})
+
     title = request.POST.get('title')
     summary = request.POST.get('summary')
     category_id = request.POST.get('category_id')
     level = request.POST.get('level')
     state = request.POST.get('state')
-    if thread_id:
 
-        try:
-            t = Thread.objects.get(id=thread_id)
-            category_id = request.POST.get('category_id')
-            if request.user.username != t.author.user.username:
-                return HttpResponseBadRequest(reason="No Edit Rights")
-
-            t.title = title
-            t.summary = summary
-            t.category_id = category_id
-            t.level = level
-            t.state = state
-            t.save()
-        except Exception as e:
-            return HttpResponseServerError(reason=str(e))
-
-        return_data = {
-            'thread_id': thread_id,
-            'title': t.title,
-            'summary': t.summary,
-            "category": {
-                "id": t.category.id,
-                "name": t.category.name
-            },
-            "level": t.level,
-            "state": t.state if t.level == "state" else "",
-            "location": t.level if not t.state else dict(US_STATES).get(t.state),
-        }
-        return JsonResponse({'data': return_data})
-    else:
-        return HttpResponseBadRequest(reason="Invalid Thread Reference")
+    try:
+        t = Thread.objects.get(id=thread_id)
+        category_id = request.POST.get('category_id')
+        if request.user.username != t.author.user.username:
+            return HttpResponseBadRequest(reason="No Edit Rights")
+
+        t.title = title
+        t.summary = summary
+        t.category_id = category_id
+        t.level = level
+        t.state = state
+        t.save()
+    except Exception as e:
+        return HttpResponseServerError(reason=str(e))
 
+    return_data = {
+        'thread_id': thread_id,
```

Comment:
Can we access the ID property directly from the `thread` object? If so, it would be consistent to do something like `thread.id` here, rather than using the variable `thread_id`. The remainder of this object uses properties of `thread` instead of placeholder variables.